src/labels.ts

```diff
@@ -34,3 +38,45 @@ const removeLabelsFromMergedPr = (labels: string[]) => {
     );
   }));
 };
+
+// for each gitea version, fetch all PRs that target that version and remove the
+// backport/* labels from them
+export const removeBackportLabelsFromPrsTargetingReleaseBranches = async () => {
+  const giteaVersions = await fetchGiteaVersions();
+  // versions
+  return Promise.all(giteaVersions.map(async (version) => {
+    const prs = await fetchTargeting(`release/v${version.majorMinorVersion}`);
+    // PRs
+    return removeBackportLabelsFromPrs(prs.items);
+  }));
+};
+
+// given a list of PRs, removes the backport/* labels from them
+export const removeBackportLabelsFromPrs = (prs) => {
+  return Promise.all(
+    prs.map((pr: {
+      title;
+      labels;
+      number: number;
+    }) =>
+      // labels
+      Promise.all(
+        pr.labels.filter((label: { name: string }) =>
+          label.name.startsWith("backport/")
```

Comment:
This double-nesting of `Promise.all` makes it hard to read. How about building a filtered array first (possibly [flatMap](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flatMap)) and then run them with a single `Promise.all`?

Reply:
Done